blocks/magazine-hero/magazine-hero.js

```diff
@@ -0,0 +1,161 @@
+import { createOptimizedPicture, decorateIcons, readBlockConfig } from '../../scripts/lib-franklin.js';
+
+/**
+ * Generic carousel block, which can be used for any content or blocks.
+ * Each row is a slide.
+ * @param block
+ */
+export default async function decorate(block) {
+  const config = readBlockConfig(block);
+  // readBlockConfig does not keep the link texts, so we store them here
+  const linkTexts = {};
+  // eslint-disable-next-line no-restricted-syntax
+  for (const a of block.querySelectorAll('a')) {
+    linkTexts[a.href] = a.textContent;
+  }
+  block.textContent = '';
```

Comment:
It would.,but then for...of is so much nicer and widespread, we should actually use it. https://caniuse.com/mdn-javascript_statements_for_of
I updated the eslint config to ignore this.